controller/action/DownloadActions.class.php

```diff
@@ -113,4 +103,34 @@ protected static function getEmailParam()
 
     return $email;
   }
+
+  public static function prepareDownloadButtonPartial(array $vars){
+       $osChoices = OS::getAll();
+
+       $os = static::guessOS();
+
+       if ($os && isset($osChoices[$os])) {
```

Comment:
This can be mildly streamlined. Rather than an `if/else` just modify `$vars` in the `if` and return `$vars` in both places.